frontend/src/views/WheresMyGene/common/store/reducer.ts

```diff
@@ -424,13 +477,87 @@ function setXAxisHeight(state: State, action: PayloadAction<number>): State {
   };
 }
 
+export interface SetFilteredCellTypesPayload {
+  filteredCellTypes: State["filteredCellTypes"];
+  filteredCellTypeIds: State["filteredCellTypeIds"];
+  displayedTissueIds: string[];
+}
+
 function setFilteredCellTypes(
   state: State,
   action: PayloadAction<SetFilteredCellTypesPayload>
 ): State {
+  const {
+    payload: { filteredCellTypes, filteredCellTypeIds },
+  } = action;
+
+  track(EVENTS.WMG_SELECT_CELL_TYPE, {
+    cell_types: filteredCellTypeIds,
+  });
+
+  return {
+    ...state,
+    filteredCellTypes,
+    filteredCellTypeIds,
+    expandedTissueIds: handleExpandedTissueIds(state, action),
+  };
+
+  function handleExpandedTissueIds(
```

Comment:
I know they're at different scopes, but having two functions in the same file with the same name threw me off

Reply:
Yeah I was debating whether to name them differently, but since they're scoped and considered "private", I thought that would be a good enough distinction. But it did throw me off as well, especially since they're both in the same reducer file

I think I'll just add a prefix to be super explicit, and when we migrate to RTK later and split reducers out to different files, I'll change the name back to `handleExpandedTissueIds()` since then we'll have file level namespace, which hopefully won't throw anyone off anymore!

Reply:
Updated!